platform/ios/export/export_plugin.cpp

```diff
@@ -282,6 +282,7 @@ void EditorExportPlatformIOS::get_export_options(List<ExportOption> *r_options)
 	r_options->push_back(ExportOption(PropertyInfo(Variant::STRING, "application/short_version", PROPERTY_HINT_PLACEHOLDER_TEXT, "Leave empty to use project version"), ""));
 	r_options->push_back(ExportOption(PropertyInfo(Variant::STRING, "application/version", PROPERTY_HINT_PLACEHOLDER_TEXT, "Leave empty to use project version"), ""));
 
+	// TODO(sgc): set to iOS 14.0 for Metal
 	r_options->push_back(ExportOption(PropertyInfo(Variant::STRING, "application/min_ios_version"), "12.0"));
```

Comment:
I can see a couple ways to do this:
1. Either check what rendering driver is configured here and register the option with a different default accordingly, but this might be a bit confusing to users to have a min value change without clear indication of why
2. Or add that check in the validation step (`has_valid_project_configuration`) and show a warning that lets users adjust the min themselves. That helps document this min requirement but isn't great UX either

Reply:
This probably should not be force, and instead we should have fallback to MoltenVK. But a warning in the export dialog should be added.

Reply:
I might need some guidance it how to resolve this one 🙇🏻

Reply:
Something like:

```diff
diff --git a/platform/ios/export/export_plugin.cpp b/platform/ios/export/export_plugin.cpp
index 8d337dbfd6..e4b5392c4e 100644
--- a/platform/ios/export/export_plugin.cpp
+++ b/platform/ios/export/export_plugin.cpp
@@ -2657,6 +2657,13 @@ bool EditorExportPlatformIOS::has_valid_export_configuration(const Ref<EditorExp
 		}
 	}
 
+	if (GLOBAL_GET("rendering/rendering_device/driver.ios") == "metal") {
+		float version = p_preset->get("application/min_ios_version").operator String().to_float();
+		if (version < 14.0) {
+			err += TTR("Metal renderer require iOS 14+.") + "\n";
+		}
+	}
+
 	if (!err.is_empty()) {
 		r_error = err;
 	}
```

Reply:
(Tweaked the message a bit to give more instructions to the user.)

The above + making the default min version 14.0 should work well, then only users who reduce this min version will see the warning.

Reply:
![CleanShot 2024-08-20 at 06 53 42@2x](https://github.com/user-attachments/assets/334d2c44-61b4-4664-bbb0-559b3cceb05d)

That was easy, thanks!